keps/sig-cli/NNNN-kustomize-components.md

```diff
@@ -0,0 +1,308 @@
+
+[kubernetes.io]: https://kubernetes.io/
+[kubernetes/enhancements]: https://git.k8s.io/enhancements
+[kubernetes/kubernetes]: https://git.k8s.io/kubernetes
+[kubernetes/website]: https://git.k8s.io/website
+
+## Summary
+
+(**FIXME:** Explain what is the current situation and then roughly explain what this feature does)
+
+> This section is incredibly important for producing high quality user-focused
+> documentation such as release notes or a development roadmap.  It should be
+> possible to collect this information before implementation begins in order to
+> avoid requiring implementors to split their attention between writing release
+> notes and implementing the feature itself.  KEP editors, SIG Docs, and SIG PM
+> should help to ensure that the tone and content of the `Summary` section is
+> useful for a wide audience.
+>
+> A good summary is probably at least a paragraph in length.
+
+## Motivation
+
+(**FIXME:** Explain that we don't want to copy-paste stuff, list issues that wanted it at some form)
+
+> This section is for explicitly listing the motivation, goals and non-goals of
+> this KEP.  Describe why the change is important and the benefits to users.  The
+> motivation section can optionally provide links to [experience reports][] to
+> demonstrate the interest in a KEP within the wider Kubernetes community.
+>
+> [experience reports]: https://github.com/golang/go/wiki/ExperienceReports
+
+### Goals
+
+(**FIXME:** Reusable configuration, anything else?)
+
+> List the specific goals of the KEP.  What is it trying to achieve?  How will we
+> know that this has succeeded?
+
+### Non-Goals
+
+(**FIXME:** What to add here?)
+
+> What is out of scope for this KEP?  Listing non-goals helps to focus discussion
+> and make progress.
+
+## Proposal
+
+(**FIXME:** High-level explanation of how it works)
+
+> This is where we get down to the specifics of what the proposal actually is.
+> This should have enough detail that reviewers can understand exactly what
+> you're proposing, but should not include things like API designs or
+> implementation.  The "Design Details" section below is for the real
+> nitty-gritty.
+
+### User Story
+
+(**FIXME:** Write here the example from the PR, minus the section for the current situation)
+
+> Detail the things that people will be able to do if this KEP is implemented.
+> Include as much detail as possible so that people can understand the "how" of
+> the system.  The goal here is to make this feel real for users without getting
+> bogged down.
+
+
+### Notes/Constraints/Caveats (optional)
+
+(**FIXME:** Write here the issue with files in the component directory)
+
+> What are the caveats to the proposal?
+> What are some important details that didn't come across above.
+> Go in to as much detail as necessary here.
+> This might be a good place to talk about core concepts and how they relate.
+
+### Risks and Mitigations
```

Comment:
This is what I propose for the **Risks and Mitigations** section:

Since a component is a special case of a kustomization, the same logic and restrictions apply here as well. This means that the threat model of components is no different from the threat model for overlays or other kustomizations.

Reply:
> Since a component is a special case of a kustomization
I would rephrase that to:

"Since components are a special case of resources that serialize operations"

Reply:
I was referring to this actually: https://github.com/kubernetes-sigs/kustomize/blob/master/docs/glossary.md#kustomization

Also, I meant "type" instead of "case".

Reply:
Ok, I'm ok with the edit.